library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInsertOutputOperator.java

```diff
@@ -146,28 +152,36 @@ private String getMatchingField(Field[] fields, String columnName)
     return null;
   }
 
-  protected void populateColumnDataTypes(String columns) throws SQLException
+  /**
+   * Function to populate Meta Data.
+   * @param columnNamesSet
+   * @throws SQLException
+   */
+  protected void populateColumnDataTypes(HashSet<String> columnNamesSet) throws SQLException
   {
-    columnNames = Lists.newArrayList();
-    columnDataTypes = Lists.newArrayList();
-    columnNullabilities = Lists.newArrayList();
-
-    try (Statement st = store.getConnection().createStatement()) {
-      if (columns == null || columns.length() == 0) {
-        columns = "*";
+    ResultSet rsColumns;
+    DatabaseMetaData meta = store.getConnection().getMetaData();
+    rsColumns = meta.getColumns(null, null, getTablename().toUpperCase(), null);
+    if (!rsColumns.isBeforeFirst()) {
+      // No Result Set, may be the table is stored in lower case
+      rsColumns = meta.getColumns(null, null, getTablename().toLowerCase(), null);
+    }
+    int totalColumns = columnNamesSet.size();
+    if (totalColumns == 0) {
+      //Direct mapping between POJO fields and DB columns
+      while (rsColumns.next()) {
+        columnNames.add(rsColumns.getString("COLUMN_NAME"));
+        columnNullabilities.add(rsColumns.getInt("NULLABLE"));
+        columnDataTypes.add(rsColumns.getInt("DATA_TYPE"));
       }
-      ResultSet rs = st.executeQuery("select " + columns + " from " + getTablename());
-
-      ResultSetMetaData rsMetaData = rs.getMetaData();
-      LOG.debug("resultSet MetaData column count {}", rsMetaData.getColumnCount());
-
-      for (int i = 1; i <= rsMetaData.getColumnCount(); i++) {
-        int type = rsMetaData.getColumnType(i);
-        String columnName = rsMetaData.getColumnName(i);
-        columnNames.add(columnName);
-        columnDataTypes.add(type);
-        columnNullabilities.add(rsMetaData.isNullable(i));
-        LOG.debug("column name {} type {}", rsMetaData.getColumnName(i), type);
+    } else {
+      while ((totalColumns > 0) && rsColumns.next()) {
+        if (columnNamesSet.contains(rsColumns.getString("COLUMN_NAME").toUpperCase())) {
+          columnNames.add(rsColumns.getString("COLUMN_NAME"));
+          columnNullabilities.add(rsColumns.getInt("NULLABLE"));
```

Comment:
NULLABLE returns 3 types of results.
https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getColumns(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String)

Can you confirm if this is inline with how columnNullabilities works?